contracts/strategy/yearn/LusdStrategy.sol

```diff
@@ -0,0 +1,170 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.10;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
+
+import {PercentMath} from "../../lib/PercentMath.sol";
+import {ERC165Query} from "../../lib/ERC165Query.sol";
+import {IStrategy} from "../IStrategy.sol";
+import {CustomErrors} from "../../interfaces/CustomErrors.sol";
+import {IYearnVault} from "../../interfaces/yearn/IYearnVault.sol";
+import {IVault} from "../../vault/IVault.sol";
+
+contract LusdStrategy is IStrategy, AccessControl, CustomErrors {
+    using SafeERC20 for IERC20;
+    using PercentMath for uint256;
+    using ERC165Query for address;
+
+    error StrategyNoUnderlying();
+    error StrategyNotEnoughShares();
+
+    bytes32 public constant MANAGER_ROLE =
+        0x241ecf16d79d0f8dbfb92cbc07fe17840425976cf0667f022fe9877caa831b08; // keccak256("MANAGER_ROLE");
+
+    IERC20 public immutable underlying;
+    /// @inheritdoc IStrategy
+    address public immutable override(IStrategy) vault;
+    // yearn vault that this strategy is interacting with
+    IYearnVault public immutable yVault;
+
+    constructor(
+        address _vault,
+        address _owner,
+        address _yVault,
+        address _underlying
+    ) {
+        if (_owner == address(0)) revert StrategyOwnerCannotBe0Address();
+        if (!_vault.doesContractImplementInterface(type(IVault).interfaceId))
+            revert StrategyNotIVault();
+
+        _setupRole(DEFAULT_ADMIN_ROLE, _owner);
+        _setupRole(MANAGER_ROLE, _vault);
+        vault = _vault;
+        yVault = IYearnVault(_yVault);
+        underlying = IERC20(_underlying);
+    }
+
+    //
+    // Modifiers
+    //
+
+    modifier onlyManager() {
+        if (!hasRole(MANAGER_ROLE, msg.sender))
+            revert StrategyCallerNotManager();
+        _;
+    }
+
+    /// @inheritdoc IStrategy
+    function hasAssets()
+        external
+        view
+        virtual
+        override(IStrategy)
+        returns (bool)
+    {
+        return _getSharesBalance() > 0;
+    }
+
+    /**
+     * Amount, expressed in the underlying currency, currently in the strategy
+     *
+     * @return The total amount of underlying invested into yearn vault
+     */
```

Comment:
I think the comments should be `/// @inheritdoc IStrategy`

Reply:
fixed